Frontend/ui-driver/src/Screens/DriverEarningsScreen/View.purs

```diff
@@ -0,0 +1,1313 @@
+          ][ imageView $
+            [ width $ V 32
+            , height $ V 32
+            , imageWithFallback $ fetchImage FF_ASSET $ if state.props.weekIndex == 3 then "ny_ic_chevron_right_light_grey" else "ny_ic_chevron_right"
+            , gravity RIGHT
+            , onClick push $ const $ RightChevronClicked state.props.weekIndex
+            , clickable $ state.props.weekIndex < 3
+            ]
+          ]
+        ]
+      , barGraphView push state
+      , separatorView true
+      , linearLayout
+        [ height WRAP_CONTENT
+        , width MATCH_PARENT
+        , gravity CENTER
+        , padding $ PaddingVertical 15 15
+        ][ linearLayout
+          [ height MATCH_PARENT
+          , gravity CENTER
+          , width $ V (((screenWidth unit) - 75)/2)
+          ][ textView $ 
+              [ height WRAP_CONTENT
+              , width WRAP_CONTENT
+              , text $ getString RIDES
+              , margin $ MarginRight 8
+              ] <> FontStyle.paragraphText TypoGraphy
+              ,  textView $ 
+              [ height WRAP_CONTENT
+              , width WRAP_CONTENT
+              , text $ show $ state.props.totalEarningsData.totalRides
+              ] <> FontStyle.h2 TypoGraphy
+          ]
+          ,linearLayout
+            [ height $ V 30
+            , width $ V 2
+            , background Color.grey900
+            ][]
+          , linearLayout
+            [ height WRAP_CONTENT
+            , gravity CENTER
+            , width $ V (((screenWidth unit) - 75)/2)
+            ][ textView $ 
+              [ height WRAP_CONTENT
+              , width WRAP_CONTENT
+              , text $ getString DISTANCE
+              , margin $ MarginRight 8
+              ] <> FontStyle.paragraphText TypoGraphy
+              ,  textView $ 
+              [ height WRAP_CONTENT
+              , width WRAP_CONTENT
+              , text $ show (state.props.totalEarningsData.totalDistanceTravelled / 1000) <> " km"
+              ] <> FontStyle.h2 TypoGraphy
+          ]
+        ] 
+      , linearLayout 
+        [ height WRAP_CONTENT
+        , width MATCH_PARENT
+        , orientation HORIZONTAL
+        , gravity CENTER
+        , margin $ MarginTop 6
+        ]( map (\index -> dotView push index state) [0,1,2,3])
+    ]
+
+dotView :: forall w . (Action -> Effect Unit) -> Int -> ST.DriverEarningsScreenState -> PrestoDOM (Effect Unit) w
+dotView push index state = 
+  linearLayout
+  [ height $ V 6
+  , width $ V 6
+  , cornerRadius 12.0
+  , background $ if index == state.props.weekIndex then Color.black800 else Color.grey900
+  , margin $ MarginHorizontal 2 2
+  ][]
+
+yatriCoinsView :: forall w . (Action -> Effect Unit) -> ST.DriverEarningsScreenState -> PrestoDOM (Effect Unit) w
+yatriCoinsView push state = 
+  linearLayout
+  [ height MATCH_PARENT
+  , width MATCH_PARENT
+  , orientation VERTICAL
+  , padding $ Padding 16 0 16 70
+  ] [  balanceView push state
+      -- , insightView push state
```

Comment:
remove commented code. If it is required then add a TODO statement with the reason for why it is required.